### PR TITLE
GLideNUI: introduce Config{Init,Cleanup}

### PR DESCRIPTION
**Pull request by Rosalie241**
This should hopefully make opening the GLideNUI window slightly faster on first launch after launching the emulator.

### CODE REVIEW

File: src/GLideNUI/GLideNUI.cpp [hunk 1]
```diff
@@ -19,29 +19,21 @@ Q_IMPORT_PLUGIN(QICOPlugin)
 inline void initMyResource() { Q_INIT_RESOURCE(icon); }
 inline void cleanMyResource() { Q_CLEANUP_RESOURCE(icon); }
 
+static std::unique_ptr<QApplication> pQApp;
+static QCoreApplication* pAppInstance;
```
Comment (gonetz) [inline src/GLideNUI/GLideNUI.cpp:23]: Please use s_ prefix for static variables: s_pQApp, s_pAppInstance

File: src/GLideNUI/GLideNUI.cpp [hunk 2]
```diff
@@ -96,6 +81,25 @@ int runAboutThread(const wchar_t * _strFileName) {
...
+		int argc = 0;
+		char * argv = 0;
+		pQApp.reset(new QApplication(argc, &argv));
+		pAppInstance = pQApp.get();
+	}
+}
+
+EXPORT void CALL ConfigCleanup(void)
+{
+	cleanMyResource();
```
Comment (gonetz) [inline src/GLideNUI/GLideNUI.cpp:100]: Application crashes on exit. now
I think, pQApp should be released here.

**Review by gonetz — CHANGES_REQUESTED** [review]

File: src/GLideNUI/GLideNUI.cpp (hunk 2 above)
Comment (Rosalie241) [reply src/GLideNUI/GLideNUI.cpp:100]: can you re-test with the latest push? I personally haven't experienced any crashes myself.

File: src/GLideNUI/GLideNUI.cpp [hunk 3]
```diff
@@ -96,6 +81,28 @@ int runAboutThread(const wchar_t * _strFileName) {
...
+		char * argv = 0;
+		s_pQApp.reset(new QApplication(argc, &argv));
+		s_pAppInstance = s_pQApp.get();
+	}
+}
+
+EXPORT void CALL ConfigCleanup(void)
+{
+	cleanMyResource();
+	if (s_pQApp) {
```
Comment (gonetz) [inline src/GLideNUI/GLideNUI.cpp:101]: Redundant check. You can safely release an empty smart pointer.

File: src/GLideNUI/GLideNUI.cpp [hunk 4]
```diff
@@ -96,6 +81,28 @@ int runAboutThread(const wchar_t * _strFileName) {
...
+		s_pQApp.release();
```
Comment (gonetz) [inline src/GLideNUI/GLideNUI.cpp:102]: Yes, this fixes the crash on exit.

**Review by gonetz — CHANGES_REQUESTED** [review]

File: src/GLideNUI/GLideNUI.cpp (hunk 3 above)
Comment (Rosalie241) [reply src/GLideNUI/GLideNUI.cpp]: Removed the check, thank you!

**Review by gonetz — APPROVED** [review]

File: src/GLideNUI/GLideNUI.cpp (hunk 2 above)
Comment (Squall-Leonhart) [reply src/GLideNUI/GLideNUI.cpp:100]: This has fixed the Pj64 plugins screen crashing too when not the selected plugin.

**Pull request merged**